src/qt/veil/veilstatusbar.cpp

```diff
@@ -59,6 +61,39 @@ void VeilStatusBar::onBtnSyncClicked(){
 
 #ifdef ENABLE_WALLET
 bool fBlockNextStakeCheckSignal = false;
+void VeilStatusBar::setStakingText() {
+
+        // Determine if staking is recently active. Note that this is not immediate effect. Staking could be disabled and it could take up to 70 seconds to update state.
+        int64_t nTimeLastHashing = 0;
+        if (!mapHashedBlocks.empty()) {
+            auto pindexBest = chainActive.Tip();
+            if (mapHashedBlocks.count(pindexBest->GetBlockHash())) {
+                nTimeLastHashing = mapHashedBlocks.at(pindexBest->GetBlockHash());
+            } else if (mapHashedBlocks.count(pindexBest->pprev->GetBlockHash())) {
+                nTimeLastHashing = mapHashedBlocks.at(pindexBest->pprev->GetBlockHash());
+            }
+        }
+        bool fStakingActive = false;
+        if (nTimeLastHashing)
+            fStakingActive = GetAdjustedTime() + MAX_FUTURE_BLOCK_TIME - nTimeLastHashing < 70;
```

Comment:
Define 70 as a static const, with a reflective name and description; then use it here and in the toast dialog further down

Reply:
I'm OK with using it within the time check.  

I suggest changing the toast to something along the lines of:  "This may take a few minutes" as
1) The display changing in the Qt is a function of when staking is active and the calls are made (there is no periodic task that performs this check, only when new blocks come in).  In my testing I've had a few minutes pass before the display is updated  though calls through the command line show active already.  This has to do with how the display is updated (when the command line is used, the calculation is done on the spot).
2) The 70 was copied from other code and I have no knowledge of what the actual high water mark is.

Reply:
The comment is merely to adjust away from using magic numbers.  Instead of "70" hardcoded in two places [the code and the message], make it 
`static const STAKING_DELAY = 70`

or something to that effect; so that it's defined; and if adjusted anytime in the future you don't have the message (stating 70 seconds) and the code diverging.

Reply:
Updated the message to be less definitive of the timeframe (as the user won't see the change real-time).  Added the constant and used it in the check.

Reply:
Did you take care of the "enabling..." thing while it's synching?

Reply:
Yup.  Works like it should now.  Have it a quick test too.